### DISCUSSION TITLE
Role dissociation

### DISCUSSION THREAD

**Discussion by edmaSB** [orchidsoftware/platform#2854] [Q&A] [answered]
I have a problem with roles. When I add or remove a permission from a role, it is removed from users
Could it be a bug?

**Comment by tabuna** [comment, marked as answer]
I'm not entirely sure I fully understand what you mean. It would be helpful if you could provide code, a screenshot, or a more detailed explanation.

I'll try to first address the possibility that you are removing the role itself, and then I'll explain that the checkbox has three states.

Imagine you are managing an organization with employees holding various positions and responsibilities. Each person has specific tasks based on their role. For instance, there are packers working in the warehouse.

When you hire a new packer, they automatically receive access to the warehouse because of their role.

However, if at some point you start receiving pre-packaged goods and eliminate the packer position, all employees who only held that role and had no other duties or special mandates will lose their access to the warehouse. 


If a user had certain permissions through a single role, and you either completely removed that role or revoked those permissions from the role, the user should also lose those permissions (unless they have them through other roles or their own mandate).

Visually, this would appear on the user's screen as follows:

1. Checkbox checked - the user has a personal mandate.
2. Dash - the user has the permission through one of their roles.
3. Empty field - the user does not have the permission.


![image](https://github.com/user-attachments/assets/c9b56b4e-6190-41e4-95bb-b5c833a96f5b)
